### PR TITLE
Payments whit no attachment - pn-8869

### PR DESCRIPTION
**Pull request by AndreaCimini90**
## Short description
Sometimes the payment attachment (no F24) is included in the act and not in the payment. So when the user select a payment, the button to download the attachment must be shown/hidden based on this condition.
We also added a delay of 1 second after the selection, before the payment is enabled

## List of changes proposed in this pull request
- Updated NotificationPaymentRecipient to manage the logic described above
- Created NotificationPaymentTitle component to have a cleaner code
- Updated the tests
- Reworked the NotificationPaymentRecipient component to reduce the complexity

## How to test
- Login with pf on dev (colombo) -> search the notification AGYX-HKJK-WGTD-202311-T-1 -> select one payment and check that the button to download the attachment is shown/hidden correctly
- Login with pf on dev (ada) -> search the notification ATMG-XVDH-WPTR-202311-T-1 -> the only payment available mustn't have the button to download the attachment
- Login with pf on dev -> search a notification with only one payment and with the attachment linked to it -> he only payment available must have the button to download the attachment

### CODE REVIEW

File: packages/pn-commons/src/components/NotificationDetail/__test__/NotificationPaymentTitle.test.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    );
+
+    expect(container).toHaveTextContent(
+      'notifiche - detail.payment.single-payment-subtitle notifiche - detail.payment.how'
+    );
+    const faq = getByTestId('faqNotificationCosts');
+    expect(faq).toBeInTheDocument();
+  });
+
+  it('render component - multi payments whit no f24', () => {
```
Comment (clombardi) [inline packages/pn-commons/src/components/NotificationDetail/__test__/NotificationPaymentTitle.test.tsx:58]: I think this test is identical to the first one.

**Review by clombardi — CHANGES_REQUESTED** [review]
The modifications are clear, and now the code is better organized besides having solved the business requirement. I just ask you to check a detail on a test.

File: packages/pn-commons/src/components/NotificationDetail/__test__/NotificationPaymentTitle.test.tsx (hunk 1 above)
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/NotificationDetail/__test__/NotificationPaymentTitle.test.tsx:58]: You are right! I have changed the first test

**Review by clombardi — APPROVED** [review]
The only point I signaled in my previous review was addressed, all OK now, ready to merge.

File: packages/pn-commons/src/components/NotificationDetail/__test__/NotificationPaymentTitle.test.tsx (hunk 1 above)
Comment (clombardi) [reply packages/pn-commons/src/components/NotificationDetail/__test__/NotificationPaymentTitle.test.tsx:58]: great

**Pull request merged**